src/controllers/credentials.ts

```diff
@@ -165,7 +180,35 @@ export class CredentialController {
 		}
 
 		const { credential, policies } = request.body;
-		const verifyStatus = request.query.verifyStatus === 'true' ? true : false;
+		const verifyStatus = request.query.verifyStatus === 'true';
+		const allowDeactivatedDid = request.query.allowDeactivatedDid === "true";
+
+		let issuerDid = "";
+		let decoded: any;
+
+		if (credential.issuer?.id) {
+			issuerDid = credential.issuer.id;
+		} else {
+			try {
+				decoded = jwt_decode(credential);
+				issuerDid = decoded.iss;
+			} catch (e) {
+				// If it's not a JWT - just skip it
+				if (!(e instanceof InvalidTokenError)) {
+					throw e;
+				}
+			}
+		}
+
+		if (!allowDeactivatedDid) {
+			const result = await new IdentityServiceStrategySetup(response.locals.customerId).agent.resolveDid(issuerDid);
+			if (!result?.didDocument || result.didDocumentMetadata.deactivated) {
```

Comment:
I think the check here should be more wider. Like what if the `result` or  `didDocumentMetadata` are empty?

Reply:
in such cases you may provide the error from `resolveDid`

Reply:
Agreed and good catch @anikitinDSR! I added similar comments above.

Reply:
Done! ✅